proof.go

```diff
@@ -453,6 +454,30 @@ func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, leavesWithoutN
 	return res
 }
 
+// ProtoToProof creates a proof from its proto representation.
+func ProtoToProof(protoProof pb.Proof) Proof {
```

Comment:
Don't we need an encoding counterpart i.e., `ProofToProto`?

Reply:
The counterpart will look like a creation of a pb struct. I think it could be done in place(but can make a counterpart).